draft-ietf-gnap-resource-servers.md

```diff
@@ -0,0 +1,502 @@
++--------+       +-------+                      +-------+
+~~~
+
+1. The client instance calls RS1 with an access token.
+
+2. RS1 presents that token to the AS to get a derived token
+    for use at RS2. RS1 indicates that it has no ability
+    to interact with the RO. Note that
+    RS1 signs its request with its own key, not the token's
+    key or the client instance's key.
+
+3. The AS returns a derived token to RS1 for use at RS2.
+
+4. RS1 calls RS2 with the token from (3).
+
+5. RS2 fulfills the call from RS1.
+
+6. RS1 fulfills the call from the original client instance.
+
+
+If the RS needs to derive a token from one presented to it, it can
+request one from the AS by making a token request as described in
+{{I-D.ietf-gnap-core-protocol}} and presenting the existing access token's
+value in the "existing_access_token" field.
+
+Since the RS is acting as a client instance, 
+the RS MUST identify itself with its own key in the `client` field and sign the
+request just as any client instance would.
+
+\[\[ [See issue #116](https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/116) \]\]
+
+~~~
+POST /tx HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Detached-JWS: ejy0...
+
+{
+    "access_token": {
+        "access": [
+            {
+                "actions": [
+                    "read",
+                    "write",
+                    "dolphin"
+                ],
+                "locations": [
+                    "https://server.example.net/",
+                    "https://resource.local/other"
+                ],
+                "datatypes": [
+                    "metadata",
+                    "images"
+                ]
+            },
+            "dolphin-metadata"
+        ]
+    },
+    "client": "7C7C4AZ9KHRS6X63AJAO",
+    "existing_access_token": "OS9M2PMHKUR64TB8N6BW7OZB8CDFONP219RP1LT0"
+}
+~~~
+
+The AS responds with a token for the downstream RS2 as described in 
+{{I-D.ietf-gnap-core-protocol}}. The downstream RS2 could
+repeat this process as necessary for calling further RS's.
+
+
+# Requesting Resources With Insufficient Access {#rs-request-without-token}
+
+If the client instance calls an RS without an access token, or with an
+invalid access token, the RS MAY respond to the client instance with an
+authentication header indicating that GNAP needs to be used
+to access the resource. The address of the GNAP
+endpoint MUST be sent in the "as_uri" parameter. The RS MAY
+additionally return a resource reference that the client instance MAY use in
+its access token request. This
+resource reference handle SHOULD be sufficient for at least the action
+the client instance was attempting to take at the RS. The RS MAY use the [dynamic resource handle request](#rs-register-resource-handle) to register a new resource handle, or use a handle that
+has been pre-configured to represent what the AS is protecting. The
+content of this handle is opaque to the RS and the client instance.
```

Comment:
"The content of this handle is opaque to the client instance" would cover the general case (dynamic or pre-configured).

Reply:
correct, this is requesting resources by reference